plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReDeployer.java

```diff
@@ -0,0 +1,471 @@
+    void getDeployRetryTimes(String localNodeId, ActionListener<SearchResponse> searchResponseActionListener) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.fetchSource(new String[] { MODEL_DEPLOY_RETRY_TIMES_FIELD }, null);
+        QueryBuilder queryBuilder = QueryBuilders.idsQuery().addIds(localNodeId);
+        searchSourceBuilder.query(queryBuilder);
+
+        SearchRequestBuilder searchRequestBuilder = new SearchRequestBuilder(client, SearchAction.INSTANCE)
+            .setIndices(ML_MODEL_REDEPLOY_INDEX)
+            .setSource(searchSourceBuilder);
+
+        searchRequestBuilder.execute(ActionListener.wrap(searchResponse -> {
+            // ML_MODEL_REDEPLOY_INDEX did not exist, it means it is our first time to do model auto-redeploying operation
+            if (searchResponse == null) {
+                searchResponseActionListener.onResponse(null);
+                return;
+            }
+
+            SearchHit[] hits = searchResponse.getHits().getHits();
+            if (CollectionUtils.isEmpty(hits)) {
+                searchResponseActionListener.onResponse(null);
+                return;
+            }
+
+            searchResponseActionListener.onResponse(searchResponse);
+        }, searchResponseActionListener::onFailure));
+    }
+
+    /**
+     * save retry times
+     *
+     * @param localNodeId node id
+     * @param retryTimes actual retry times
+     * @param indexResponseActionListener the listener to handle the response
+     */
+    @VisibleForTesting
+    void saveLatestDeployRetryTimes(String localNodeId, int retryTimes, ActionListener<IndexResponse> indexResponseActionListener) {
+        Map<String, Object> content = new HashMap<>(2);
+        content.put(NODE_ID_FIELD, localNodeId);
+        content.put(MODEL_DEPLOY_RETRY_TIMES_FIELD, retryTimes);
+
+        IndexRequestBuilder indexRequestBuilder = new IndexRequestBuilder(client, IndexAction.INSTANCE, ML_MODEL_REDEPLOY_INDEX)
+            .setId(localNodeId)
+            .setSource(content)
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+        indexRequestBuilder.execute(ActionListener.wrap(indexResponse -> {
+            if (indexResponse.status() == RestStatus.CREATED || indexResponse.status() == RestStatus.OK) {
+                log.info("node id:{} insert deploy retry times successfully, retryTimes is {}", localNodeId, retryTimes);
+                indexResponseActionListener.onResponse(indexResponse);
+            }
+        }, e -> {
+            log.error("node id:{} insert deploy retry times unsuccessfully", localNodeId, e);
+            indexResponseActionListener.onFailure(new MLException(e));
+        }));
+    }
+
+    /**
+      * check whether the local node is ml node
+     *
+     * @param roleName role name
+    * @return if the local node is ml node,return true,else return false
+     */
+    @VisibleForTesting
+    boolean isNode(String roleName) {
```

Comment:
How about change the function name to `hasRoleName`?

Reply:
changed